src/aims_ui/models/address.py

```diff
@@ -0,0 +1,52 @@
+import json
+
+class AddressAttribute():
+  def __init__(self, address_data, name, ):
+    self.name = name
+    self.raw_value = address_data.get(name)
+    self.value = self.format_special(self.raw_value)
+    self.show= False
+
+    # Set values to show in small overview of an address
+    values_to_show = ['uprn',
+        'classificationCode',
+        'confidenceScore',
+        ]
+
+    if name in values_to_show:
+      self.show = True
+
+  def format_special(self, value):
+    # Special formatting for some values
+    if self.name == 'confidenceScore':
+      return( f'{value}% match' )
+    if self.name == 'geo':
+      # Convert geo to strings
+      new_d = {'longitude':str( value.get('longitude')),
+               'latitude': str(value.get('latitude')),}
+      return new_d 
+
+    return f'{value}'
+    
+
+
+class Address():
+  def __init__(self, address_data):
+    # Essentially all atributes of an expected address from AIMS API
+    self.uprn=AddressAttribute(address_data, 'uprn')
+    self.formattedAddress = AddressAttribute(address_data, 'formattedAddress')
```

Comment:
Generally, Python doesn't use camelCase for values. I'm unsure if the names of these are used later on.

The way this is constructed seems a bit odd- it's possible a dataclass could provide some useful helpers.

Hardcoding the way formatting works by using the attribute name is a little unclear. Using very simplistic dataclasses for a CondidenceScore and Geo dataclass may be more Pythonic.

Reply:
In this case I've used the case style to match the api, as the api returns these values in camelCase. I thought this would be the better option for consistency through both products

Reply:
For JSON input/output and other data in transit, or displayed on-screen, camelCase would be best, yes. It's more to do with the internal names of the variables, and keeping a consistent code style inside the product, so developers don't have to guess which classes use camelCase and which use snake_case for every single data class.